tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,218 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run open-vm-tools testing against VMware ESXi
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_autotest::esxi_utils;
+use Time::Local;
+
+sub run {
+    assert_script_run('rpmquery open-vm-tools');
+    systemctl('status vmtoolsd');
+    systemctl('status vgauthd');
+
+    my $vm_name = console('svirt')->name;
+    record_info('Guest Name', $vm_name);
+    my $vm_id = esxi_vm_get_vmid($vm_name);
+    record_info('Guest ID', $vm_id);
+    my $vm_ip = esxi_vm_public_ip($vm_id);
+    record_info('Guest IP', $vm_ip);
+
+    record_info('Power Manangement Tests');
+    power_management_tests($vm_name, $vm_id, $vm_ip);
+    record_info('Networking Tests');
+    networking_tests($vm_id, $vm_ip);
+    record_info('Clock Sync Tests');
+    clock_sync_tests($vm_name, $vm_id, $vm_ip);
+}
+
+sub get_timelc {
+    my ($time, $timelc, $sec, $min, $hour, $mday, $mon, $year);
+    $time = $_[0];
+
+    my @time = reverse(split /-|:|\s|\//, $time);
+    foreach (@time) {
+        s/^0//;
+        $timelc .= $_ . ",";
+    }
+    $timelc =~ s/,$//;
+    ($sec, $min, $hour, $mday, $mon, $year) = split /,/, $timelc;
+    $year -= 1900;
+    $mon  -= 1;
+    $timelc = timelocal($sec, $min, $hour, $mday, $mon, $year);
+
+    return $timelc;
+}
+
+sub power_management_tests {
```

Comment:
For sub/function name, it is better to be an action, rather than a noun, eg this sub can be named as do_power_management_tests. So does others.

Reply:
OK, adding prefix "do_" to all test functions.